hyperspy/misc/utils.py

```diff
@@ -1007,8 +1010,25 @@ def stack(signal_list, axis=None, new_axis_name="stack_element", lazy=None,
         newlist = broadcast_signals(*signal_list, ignore_axis=axis_input)
 
         if axis is not None:
-            step_sizes = [s.axes_manager[axis].size for s in newlist]
-            axis = newlist[0].axes_manager[axis]
+            step_sizes = [s.axes_manager[axis_input].size for s in newlist]
+            axis = newlist[0].axes_manager[axis_input]
+            if not axis.axes_manager[axis_input].is_uniform:
+                # stack axes if non-uniform
+                if hasattr(axis.axes_manager[axis_input],'_expression'):
```

Comment:
Is it more explicit/clear to check if the type of the axis is FunctionalDataAxis?

Reply:
I could check the value of the `__class__` attribute to be explicit, but `_expression` is unique as well. But you are right that it might be clearer to others that way.

Reply:
I was thinking `if type(axis.axes_manager[axis_input]) is FunctionalDataAxis`.

I'm also a little confused on how the `axis` variable changes in this function - initially its an int / string, but then `newlist[0].axes_manager[axis_input]` gets returned into it - is `axis` now the DataAxis, and `axis.axes_manager[axis_input] == axis`?

If so, my previous if statement gets reduced to `if type(axis) is FunctionalDataAxis`.

Reply:
Arguably, `if axis.is_uniform` could just be `if type(axis) is UniformDataAxis`, and we could do away with the `axis.is_uniform` and `axis._is_uniform` bits.